tests/unit/test_utils.py

```diff
@@ -352,6 +354,36 @@ def test_unpack_zip(self, data):
         unzip_file(test_file, self.tempdir)
         self.confirm_files()
 
+    def test_unpack_zip_failure(self):
+        """
+        Test unpacking a *.zip with file containing .. path
+        and expect exception
+        """
+        test_zip = os.path.join(self.tempdir, 'test_zip.zip')
+        with zipfile.ZipFile(test_zip, 'w') as myzip:
+            myzip.writestr(os.path.join('..', 'outside_file.txt'),
+                           'outside file')
+            myzip.writestr('regular_file.txt', 'regular file')
```

Comment:
I would make a helper function `make_test_zip_file()` for the above lines that accepts a list of paths so in the test method the lines above could be replaced by `make_test_zip_file(test_zip, ['regular_file.txt', '../outside_file.txt'])`.

Reply:
creating a helper function is a little bit over engineered i think unless we would be adding more files to test but for now, just the current files would be sufficient, what do you think

Reply:
The helper function will help for the success case I’m asking you to do. The only difference between the two tests (success case and failing case) will be the list of paths passed to the helper function.